### PR TITLE
Implement product operator and DIV_SCALAR_ARRAY in Cpp runtime

### PR DESCRIPTION
**Pull request by rfranke**
This was missing so far.

### CODE REVIEW

File: OMCompiler/SimulationRuntime/cpp/Core/Math/ArrayOperations.cpp [hunk 1]
```diff
@@ -492,6 +492,17 @@ T sum_array (const BaseArray<T>& x)
...
 }
 
+template <typename T>
+T product_array(const BaseArray<T>& x)
+{
+  const T* data = x.getData();
+  size_t nelems = x.getNumElems();
+  T val = 1;
+  for (size_t i = 0; i < nelems; i++)
+    val *= *data++;
```
Comment (phannebohm) [inline OMCompiler/SimulationRuntime/cpp/Core/Math/ArrayOperations.cpp:502]: This should be more in line with `sum_array`.
```suggestion
  T val = std::accumulate(data, data + x.getNumElems(), T(1), std::multiplies<T>());
```

**Review by phannebohm — APPROVED** [review]

**Comment by rfranke** [comment]
@phannebohm: thank you for the improved formulation

**Comment by phannebohm** [comment]
Oops, now the product is wrong. Did I do that?

**Comment by rfranke** [comment]
It's strange. `f4` makes sure that `product` isn't pre-evaluated. Running
```
   omc arrayOperationsTest.mos
```
logs the correct result `f4 = product{1, 2}) == 2`.
```
  rtest arrayOperationsTest.mos
```
obtains the wrong result `1` though?!

**Comment by perost** [comment]
> It's strange. `f4` makes sure that `product` isn't pre-evaluated. Running
> 
> ```
>    omc arrayOperationsTest.mos
> ```
> 
> logs the correct result `f4 = product({1, 2}) == 2`.
> 
> ```
>   rtest arrayOperationsTest.mos
> ```
> 
> obtains the wrong result `1` though?!

The test case has `cflags: -d=-newInst`, so it's run with the old frontend if you use rtest. And the old frontend flattens `f4` to:
```Modelica
function ArrayOperations.f4
  input Real[:] v;
  output Real p;
algorithm
  p := 1.0;
end ArrayOperations.f4;
```
which is obviously completely wrong. I suggest just removing the flag and using the new frontend instead.

**Comment by rfranke** [comment]
Better not use the comment
```
// cflags: -d=-newInst
```
but state
```
setCommandLineOptions("-d=-newInst"); getErrorString();
```
if needed. This is clearer works with omc as well.

Btw. I added a second missing operation to this pull request to reduce merge conflicts for multiple issues. This PR can still be squashed as the two extensions don't coincide.

**Pull request merged**